contrib/rocksdb_logtool.py

```diff
@@ -0,0 +1,198 @@
+        joshua._insert_blob(joshua.db, subspace, archive, offset=0)
+
+
+def download_logs(ensemble_id: str, test_uid: str):
+    with tempfile.NamedTemporaryFile() as archive:
+        subspace = _get_log_subspace(ensemble_id, test_uid)
+        logger.debug(
+            f"Downloading the archive to {archive.name} at subspace {subspace}"
+        )
+        joshua._read_blob(joshua.db, subspace, archive)
+        logger.debug(f"Tarfile size: {os.path.getsize(archive.name)}")
+        _tar_extract(archive.name)
+
+
+def list_commands(ensemble_id: str):
+    for item in joshua.tail_results(ensemble_id, errors_only=True):
+        test_harness_output = item[4]
+        match = TEST_UID_REGEXP.search(test_harness_output)
+        if not match:
+            logger.warning(f"Test UID not found in {test_harness_output}")
+            continue
+        test_uid = match.groupdict()["uid"]
+        print(f"python3 {__file__} download --ensemble-id {ensemble_id} --test-uid {test_uid}")
+
+
+def _setup_args():
+    parser = argparse.ArgumentParser(prog="rocksdb_logtool.py")
+
+    parser.add_argument(
+        "--cluster-file", type=str, default=None, help="Joshua FDB cluster file"
+    )
+
+    subparsers = parser.add_subparsers(help="Possible actions", dest="action")
+
+    report_parser = subparsers.add_parser(
+        "report", help="Check the log file, upload them to Joshua cluster if necessary"
+    )
+    report_parser.add_argument(
+        "--work-directory", type=str, default=os.getcwd(), help="Work directory"
+    )
+    report_parser.add_argument(
+        "--log-directory",
+        type=str,
+        required=True,
+        help="Directory contains XML/JSON logs",
+    )
+    report_parser.add_argument(
+        "--ensemble-id", type=str, default=None, required=False, help="Ensemble ID"
+    )
+    report_parser.add_argument("--test-uid", type=str, required=True, help="Test UID")
+
+    download_parser = subparsers.add_parser(
+        "download", help="Download the log file from Joshua to local directory"
+    )
+    download_parser.add_argument(
+        "--ensemble-id", type=str, required=True, help="Joshua ensemble ID"
+    )
+    download_parser.add_argument("--test-uid", type=str, required=True, help="Test UID")
+
+    list_parser = subparsers.add_parser(
+        "list",
+        help="List the possible download commands for failed tests in a given ensemble. NOTE: It is possible that the test is not relevant to RocksDB and no log file is available. It is the user's responsibility to verify if this happens.",
+    )
+    list_parser.add_argument(
+        "--ensemble-id", type=str, required=True, help="Joshua ensemble ID"
+    )
+
+    return parser.parse_args()
+
+
+def _main():
+    args = _setup_args()
+
+    logging.basicConfig(level=logging.INFO)
+
+    logger.debug(f"Using cluster file {args.cluster_file}")
+    joshua.open(args.cluster_file)
+
+    if args.action == "report":
```

Comment:
Maybe change "report" to "upload", which is more accurate of what the action is?